### ISSUE TITLE
Failed to load plugin in monorepo

### ISSUE THREAD

**Issue by dev-justtrade** [sindresorhus/eslint-plugin-unicorn#1871]
ESLint with plugin unicorn brakes in a NX monorepo when having packages on a deeper folder level.

It works fine in packages like `/libs/first-level` (path `@nx-monorepo/first-level`) but throws this error when used in packages like `/libs/first-level/second-level` (path `@nx-monorepo/first-level/second-level`):
```
Error: Failed to load plugin 'unicorn' declared in '.eslintrc.json » ../../../.eslintrc.json': Invalid name: "@nx-monorepo/first-level/second-level"

Error: Failed to load plugin 'unicorn' declared in '.eslintrc.json » ../../../.eslintrc.json': Invalid name: "@nx-monorepo/first-level/second-level"
    at ensureValidName (/home/xxx/nx-monorepo/node_modules/normalize-package-data/lib/fixer.js:336:15)
    at Object.fixNameField (/home/xxx/nx-monorepo/node_modules/normalize-package-data/lib/fixer.js:215:5)
    at /home/xxx/nx-monorepo/node_modules/normalize-package-data/lib/normalize.js:32:38
    at Array.forEach (<anonymous>)
    at normalize (/home/xxx/nx-monorepo/node_modules/normalize-package-data/lib/normalize.js:31:15)
    at module.exports.sync (/home/xxx/nx-monorepo/node_modules/read-pkg/index.js:37:36)
    at module.exports.sync (/home/xxx/nx-monorepo/node_modules/read-pkg-up/index.js:27:24)
    at Object.<anonymous> (/home/xxx/nx-monorepo/node_modules/eslint-plugin-unicorn/rules/expiring-todo-comments.js:50:33)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
```

I seems to be related to the `expiring-todo-comments` rule.

**Comment by schmkr** [comment]
Also experiencing this issue, in an NX monorepo.

Would it be an idea, to add a configuration option for the location of the `package.json` to use? Similar to how `import/no-extraneous-dependencies` [does this](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-extraneous-dependencies.js)?

**Comment by schmkr** [comment]
@fregante the issue is specific to a dependency unicorn uses, because the `name` property automatically generated by Nx are considered invalid according to the `normalize-package-data` package (which is a transitive dependency of this project). Those package.json files are only meant for internal use, they are not plubished to an NPM registry, but consumed as a locally built package and the package names correspond to their mapping in Typescript. Something Nx refers to as "[buildable libraries](https://nx.dev/more-concepts/buildable-and-publishable-libraries#buildable-libraries)".

The issue happens in the `rules/expiring-todo-comments.js` as a result of this block:

```js
const packageResult = readPkgUp.sync();
const hasPackage = Boolean(packageResult);
const packageJson = hasPackage ? packageResult.packageJson : {};
```

Changing that first line to `const packageResult = readPkgUp.sync({ normalize: false });` will skip validation of the package.json data and the error disappears. Reading up on what the [normalization all validates](https://github.com/npm/normalize-package-data#what-normalization-currently-entails) and comparing that with the what data is consumed (only the list of dependencies and devDependencies), to me this seems like the normalization is not really needed. 

Would you agree?

**Comment by schmkr** [comment]
FYI I have created a PR (#2020) with the above suggested change.

**Comment by sindresorhus** [comment]
Closing as it's not a problem with this plugin.
